cmd/state-svc/test/integration/svc_int_test.go

```diff
@@ -249,6 +253,72 @@ func (suite *SvcIntegrationTestSuite) checkEnabled(exec string, opts autostart.O
 	}
 }
 
+func (suite *SvcIntegrationTestSuite) TestLogRotation() {
+	suite.OnlyRunForTags(tagsuite.Service)
+	ts := e2e.New(suite.T(), true)
+	defer ts.Close()
+
+	logDir := filepath.Join(ts.Dirs.Config, "logs")
+
+	// Create a tranche of 30-day old dummy log files.
+	numFooFiles := 50
+	thirtyDaysOld := time.Now().Add(-24 * time.Hour * 30)
+	for i := 1; i <= numFooFiles; i++ {
+		logFile := filepath.Join(logDir, fmt.Sprintf("foo-%d%s", i, logging.FileNameSuffix))
+		err := fileutils.Touch(logFile)
+		suite.Require().NoError(err, "could not create dummy log file")
+		err = os.Chtimes(logFile, thirtyDaysOld, thirtyDaysOld)
+		suite.Require().NoError(err, "must be able to change file modification times")
+	}
+
+	// Override state-svc log duration timer duration from 1 minute to 10 seconds for this test.
+	logRotateTimerDuration := 10
```

Comment:
Can we bring this down as much as possible? This adds 10 seconds to the test, which is pretty significant.

Reply:
I'll try bringing it down to 4 seconds, but intermittent CI slowness may cause more nightly test failure troubles.